src/testRunner/externalCompileRunner.ts

```diff
@@ -67,7 +67,7 @@ namespace Harness {
                         cwd = config.path ? path.join(cwd, config.path) : submoduleDir;
                     }
                     if (fs.existsSync(path.join(cwd, "package.json"))) {
-                        if (fs.existsSync(path.join(cwd, "package-lock.json"))) {
+                        if (process.env.TRAVIS_EVENT_TYPE === "cron" && fs.existsSync(path.join(cwd, "package-lock.json"))) {

```

Comment:
Just leaving the lockfile in place wouldn't be enough to make it get used, it'd also need to be committed (otherwise no PR build is going to have them anyway, as it's always a fresh clone for a CI build), and installation would need to use `npm ci` rather than `npm i` (since `npm i` updates the lockfile by default). Then, for this whole scheme to be usable, we'd need to ensure that the `package-lock.json` updates are included in and themselves create a "user test baselines updated" PR (meaning we'd need to "fail the build" and open a PR if any package locks update). 

Oh, and all this is only actually useful if someone actually _reviews_ those `package-lock` updates and understands them and the effects they have on the associated builds, rather than always just checkbox-merging them; otherwise we may as well just not have a lockfile, since we're essentially blindly installing the latest allowable packages anyway, and we may as well cut the middlemen.

Reply:
Yes to everything you've said. Would it be possible to include `package-lock.json` files in the "User test baselines have changed" PRs, but continue to create a PR only if the baselines themselves have changed?

I.e. add them to the existing "User test baselines have changed" PRs, but don't create a PR/fail the build for `package-lock.json`-only changes?